.github/copilot-instructions.md

```diff
@@ -0,0 +1,235 @@
+# EICrecon Development Instructions
+
+EICrecon is a JANA-based reconstruction software for the ePIC detector. This is a complex high-energy physics reconstruction software requiring specialized dependencies and extensive build times.
+
+**ALWAYS follow these instructions first and fallback to additional search and context gathering only if the information in these instructions is incomplete or found to be in error.**
+
+## Working Effectively with EICrecon
+
+### Essential Setup: Use eic-shell Environment
+
+**CRITICAL: This is the ONLY recommended approach for development. Manual dependency installation is complex and strongly discouraged.**
+
+**Bootstrap the eic-shell environment:**
+```bash
+mkdir ~/eic
+cd ~/eic
+curl --location https://get.epic-eic.org | bash
+./eic-shell
+```
+
+**Alternative if /cvmfs is available:**
+```bash
+# Note: On JLab ifarm, run 'module load singularity/3.9.5' first
+singularity exec /cvmfs/singularity.opensciencegrid.org/eicweb/eic_xl:nightly eic-shell
+```
+
```

Comment:
````suggestion

This approach is achieved by the following GitHub Actions snippet (inside :
```yaml
    steps:
    - name: Ensure CernVM-FS is available
      uses: cvmfs-contrib/github-action-cvmfs@v5
    - name: Run inside the eic_xl:nightly environment
      uses: eic/run-cvmfs-osg-eic-shell@main
      with:
        platform-release: "eic_xl:nightly"
        run: |
          echo
```

````